src/workerd/io/trace.h

```diff
@@ -428,12 +428,17 @@ class SpanBuilder {
 
 public:
   explicit SpanBuilder(kj::Maybe<kj::Own<SpanObserver>> observer, kj::StringPtr operationName,
-                       kj::Date startTime = kj::systemPreciseCalendarClock().now())
-      { KJ_IF_MAYBE(o, observer) { state.emplace(kj::mv(*o), operationName, startTime); } }
+                       kj::Date startTime = kj::systemPreciseCalendarClock().now()) {
+    if (observer != nullptr) {
+      observer = kj::mv(observer);
```

Comment:
This line appears to assign `observer` to itself, so has no effect... do we need more testing?